cogs/mod.py

```diff
@@ -1516,10 +1516,14 @@ def format_case_msg(self, case):
         if self.settings[server.id]["delete_repeats"]:
             if not message.content:
                 return False
-            self.cache[author].append(message)
-            msgs = self.cache[author]
-            if len(msgs) == 3 and \
-                    msgs[0].content == msgs[1].content == msgs[2].content:
+            if author.id not in self.cache:
+                self.cache[author.id] = deque(maxlen=3)
+            if len(self.cache) > 100000:
```

Comment:
I suppose this could be a `while` instead of `if`, in case the length of the cache somehow becomes greater than 100001, however I don't see that happening unless it's changed outside the cog, so it's not that big of a deal

Reply:
Yeah I had the same train of thought. I'll swap it with `while` anyway and move the `move_to_end` before the deletion to be extra-safe.